crates/pallets/starknet/src/simulations.rs

```diff
@@ -98,55 +102,57 @@ impl<T: Config> Pallet<T> {
 
                 let result = res.0.map_err(|e| {
                     log::error!("Transaction execution failed during simulation: {e}");
-                    PlaceHolderErrorTypeForFailedStarknetExecution
+                    Error::from(e)
                 });
 
                 Ok((res.1, result))
             })
-            .collect::<Result<Vec<_>, PlaceHolderErrorTypeForFailedStarknetExecution>>()
-            .map_err(|_| Error::<T>::TransactionExecutionFailed)?;
+            .collect::<Result<Vec<_>, Error>>()?;
 
         Ok(tx_execution_results)
     }
 
     pub fn simulate_message(
         message: L1HandlerTransaction,
         simulation_flags: &SimulationFlags,
-    ) -> Result<Result<TransactionExecutionInfo, PlaceHolderErrorTypeForFailedStarknetExecution>, DispatchError> {
+    ) -> Result<Result<TransactionExecutionInfo, Error>, Error> {
         storage::transactional::with_transaction(|| {
             storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(Self::simulate_message_inner(
                 message,
                 simulation_flags,
             )))
         })
-        .map_err(|_| Error::<T>::FailedToCreateATransactionalStorageExecution)?
+        .map_err(|_| Error::FailedToCreateATransactionalStorageExecution)?
     }
 
     fn simulate_message_inner(
         message: L1HandlerTransaction,
         _simulation_flags: &SimulationFlags,
-    ) -> Result<Result<TransactionExecutionInfo, PlaceHolderErrorTypeForFailedStarknetExecution>, DispatchError> {
+    ) -> Result<Result<TransactionExecutionInfo, Error>, Error> {
         let block_context = Self::get_block_context();
         let mut state = BlockifierStateAdapter::<T>::default();
 
         let tx_execution_result = Self::execute_message(&message, &mut state, &block_context).map_err(|e| {
             log::error!("Transaction execution failed during simulation: {e}");
-            PlaceHolderErrorTypeForFailedStarknetExecution
+            Error::from(e)
         });
 
         Ok(tx_execution_result)
     }
 
-    pub fn estimate_message_fee(message: L1HandlerTransaction) -> Result<(u128, u128, u128), DispatchError> {
+    pub fn estimate_message_fee(message: L1HandlerTransaction) -> Result<(u128, u128, u128), Error> {
+        let mut res = None;
+
         storage::transactional::with_transaction(|| {
-            storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(Self::estimate_message_fee_inner(
-                message,
-            )))
+            res = Some(Self::estimate_message_fee_inner(message));
+            storage::TransactionOutcome::Rollback(Result::<_, DispatchError>::Ok(()))
         })
-        .map_err(|_| Error::<T>::FailedToCreateATransactionalStorageExecution)?
+        .map_err(|_| Error::FailedToCreateATransactionalStorageExecution)?;
+
+        res.expect("`res` should have been set to `Some` at this point")
```

Comment:
let's throw an Error instead of `expect`

Reply:
YOu don't have to use an outer variable, you can use the `with_transactional` return, can't you?